clients/src/test/java/org/apache/kafka/common/network/SSLSelectorTest.java

```diff
@@ -218,6 +222,49 @@ public void testRenegotiation() throws Exception {
         }
     }
 
+
+    /**
+     * Tests handling of BUFFER_UNDERFLOW during unwrap when network read buffer is smaller than SSL session packet buffer size.
+     */
+    @Test
+    public void testNetReadBufferResize() throws Exception {
+        String node = "0";
+        createSelector(10, 0, 0);
```

Comment:
I think it's a bit brittle that the new tests create their own selector and assign it to the `selector` field while the previous ones do not. So, if the ordering of the test declaration is changed, things would potentially break. Maybe all tests should create a selector at the start via a helper method?